File: app/src/main/java/com/reednit/android/repository/remote/json/LinkDataJson.java
package com.reednit.android.repository.remote.json;

import com.squareup.moshi.Json;

@SuppressWarnings("unused")
public class LinkDataJson {

    @Json(name = "name")
    public String name;
    @Json(name = "subreddit")
    public String subreddit;
    @Json(name = "title")
    public String title;
    @Json(name = "thumbnail")
    public String thumbnail;
    @Json(name = "url")
    public String url;
    @Json(name = "author")
    public String author;
    @Json(name = "created_utc")
    public long createdUtc;
    @Json(name = "is_self")
    public boolean isSelf;
    @Json(name = "is_video")
    public boolean isVideo;
    @Json(name = "selftext")
    public String selftext;
    @Json(name = "selftext_html")
    public String selftextHtml;

}

File: app/src/main/java/com/reednit/android/repository/remote/json/LinkDataJson.kt
package com.reednit.android.repository.remote.json

import com.squareup.moshi.Json

data class LinkDataJson(

        @Json(name = "name")
        val name: String,
        @Json(name = "subreddit")
        val subreddit: String,
        @Json(name = "title")
        val title: String,
        @Json(name = "thumbnail")
        val thumbnail: String,
        @Json(name = "url")
        val url: String,
        @Json(name = "author")
        val author: String,
        @Json(name = "created_utc")
        val createdUtc: Long,
        @Json(name = "is_self")
        val isSelf: Boolean,
        @Json(name = "is_video")
        val isVideo: Boolean,
        @Json(name = "selftext")
        val selftext: String,
        @Json(name = "selftext_html")
        val selftextHtml: String

)